CakeScripts.Tests/VersionParsersTests.cs

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Charlie Poole, Rob Prouse and Contributors. MIT License - see LICENSE.txt
+
+namespace CakeScripts.Tests;
+
+public class CakeTests
+{
+    [TestCase("1.2.3-beta.1", "1.2.3.1")]
+    [TestCase("1.2.3", "1.2.3.0")]
+    public void ThatWeCanParseFileVersionStrings(string versionString, string expected)
+    {
+        
+        var parsedVersion = VersionParsers.ParseFileVersion(versionString);
+        Assert.That(parsedVersion, Is.EqualTo(expected));
+    }
+
+    [TestCase("1.2.3-beta.1", "1.2.3.0")]
+    [TestCase("1.2.3", "1.2.3.0")]
+    public void ThatWeCanParseAssemblyVersionStrings(string versionString, string expected)
+    {
+        var parsedVersion = VersionParsers.ParseAssemblyVersion(versionString);
+        Assert.That(parsedVersion, Is.EqualTo(expected));
+    }
```

Comment:
I think that you have `Assembly` and `File` version the wrong way around.
As per [documentation](https://learn.microsoft.com/en-us/troubleshoot/developer/dotnet/framework/general/assembly-version-assembly-file-version#assemblyversion) the `AssemblyVersion` must be unique

Reply:
@manfred-brands The AssemblyVersion should indicate the Major.Minor version (above also patch in included, but we're not quite semver yet), and not change for each build or revision.   The FileVersion (aka AssemblyFileVersion) should change for all patches and revisons.
To me it looks like that is what the code above do.
Or do you see something I don't (not surprised though).

Reply:
The code does what you say it does. But my interpretation of the linked documentation is that the AssembyVersion should be unique so the .NET Resolver find the correct instance.
If one builds against 1.2.3-beta.1 then later against 1.2.3-beta.2 which has a bugfix, how would the compiler (and developer) know that we have the correct version if both are versioned as 1.2.3.0.
Maybe one of our native English (@stevenaw ?) speakers can read the linked documentation.

Reply:
>  In central or final release builds it makes more sense to change the AssemblyVersion and most keep the AssemblyFileVersion same as assembly version.

Thanks @CharliePoole !  I then think this is the appropriate solution for us.

And Charlie - this does fix the issue.   The issue arised because we had to manually set these numbers, and that was forgotten.  Now they are automatically set together with the package version, and will follow the package version, given we now have a resolution for how these numbers should be :-)

Reply:
This is actually not that easy.  We want the AssemblyVersion to be (equal to or) higher than the beta releases.  If we add the beta number as the 4th digit, that will be 0 for release, which means lower than the beta ones.
Earlier when we just did this manually, we kept the File and Assembly version numbers equal and fixed through the beta period.  
<img width="188" alt="image" src="https://github.com/nunit/nunit/assets/203432/123b1f87-e042-4980-bc6c-b4e2f57e2e54">
and
<img width="167" alt="image" src="https://github.com/nunit/nunit/assets/203432/90eae32d-0948-495b-968a-92e5b7539883">

So if we continue to follow this practice we would not create anything more than we potentially have had for years.